lib/money/currency.rb

```diff
@@ -243,12 +264,11 @@ def stringify_keys
     # @example
     #   Money::Currency.new(:usd) #=> #<Money::Currency id: usd ...>
     def initialize(id)
-      id = id.to_s.downcase
-      unless self.class.stringified_keys.include?(id)
-        raise UnknownCurrency, "Unknown currency '#{id}'"
-      end
       @id = id.to_sym
       initialize_data!
+      self.class.instances_mutex.synchronize do
+        self.class.instances[id] = self
+      end
```

Comment:
Also found a `thread_safe` gem useful in case, but if are you ok with few instances, then i will remove tests for thread safety at all